custom-matchers/contracts/AssetMatcherCollection.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+import "@rarible/exchange-interfaces/contracts/IAssetMatcher.sol";
+
+/*
+ * Custom matcher for collection (assetClass, that need any/all elements from collection)
+ */
+contract AssetMatcherCollection is IAssetMatcher {
+
+    bytes constant EMPTY = "";
+
+    function matchAssets(LibAsset.AssetType memory leftAssetType, LibAsset.AssetType memory rightAssetType) public pure override returns (LibAsset.AssetType memory) {
+        if ((rightAssetType.assetClass == LibAsset.ERC721_ASSET_CLASS) || (rightAssetType.assetClass == LibAsset.ERC1155_ASSET_CLASS)) {
```

Comment:
what about lazy minted items? can this be included here too? also, can lazy and not lazy assets match? Discuss this with @disenotov

Reply:
I think we need to add checking for Lazy Items  here
But  I need to write tests for  lazy and not lazy assets match in separate task RPC-105